gateway/src/apicast/proxy.lua

```diff
@@ -313,65 +309,64 @@ local function response_codes_data()
   return params
 end
 
-local function post_action(_, self, cached_key, backend, ...)
+local function handle_timer_post_action(_, self, semaphore, cached_key, backend, ...)
   local res = util.timer('backend post_action', backend.authrep, backend, ...)
 
-  if not using_post_action then
-    timers:post(1)
+  if not using_sync_post_action then
+    semaphore:post(1)
```

Comment:
I think this is really important when designing the `resty.concurrent.task`.
It needs to release the semaphore before continuing with the handler (that might terminate the request and the processing with `ngx.exit`) (even though I'm not sure how that behaves when executed in a timer)